### PR TITLE
core(fr): extract warnings from gather-runner

### PR DESCRIPTION
**Pull request by patrickhulce**
**Summary**
The 2nd to last major piece of gather runner shared logic (base artifacts left) 🎉 

**Related Issues/PRs**
ref #11313

### CODE REVIEW

File: lighthouse-core/gather/driver/navigation.js [hunk 1]
```diff
@@ -58,7 +79,7 @@ function resolveWaitForFullyLoadedOptions(options) {
  * @param {LH.Gatherer.FRTransitionalDriver} driver
  * @param {string} url
  * @param {NavigationOptions} options
- * @return {Promise<{finalUrl: string, timedOut: boolean}>}
+ * @return {Promise<{finalUrl: string, timedOut: boolean, warnings: Array<LH.IcuMessage>}>}
```
Comment (connorjclark) [inline lighthouse-core/gather/driver/navigation.js:82]: do we still need `timedOut`?

Comment (patrickhulce) [reply lighthouse-core/gather/driver/navigation.js:82]: not to accomplish today's functionality, no. prefer removal?

Comment (connorjclark) [reply lighthouse-core/gather/driver/navigation.js:82]: ya

**Review by connorjclark — APPROVED** [review]

File: lighthouse-core/gather/driver/environment.js [hunk 2]
```diff
@@ -6,7 +6,28 @@
 'use strict';
 
 const log = require('lighthouse-logger');
+const constants = require('../../config/constants.js');
 const pageFunctions = require('../../lib/page-functions.js');
```
Comment (oxd7) [inline lighthouse-core/gather/driver/environment.js:10]: https://github.com/GoogleChrome/lighthouse/pull/12469/commits/0d62b38a2d4206119e8bb3bed893fa3af66c134b#diff-048b9f3fa7dad64326d398309830c72d43e9ef97e8b45f8528e3930f0787224c

**Pull request merged**